### PR TITLE
Fix/HNJ-16 improve pre push hook

### PR DESCRIPTION
**Pull request by zarinaazamatova**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,12 +43,13 @@
     "lint-staged": "^13.2.2",
     "prettier": "^2.8.8",
     "ts-jest": "^29.1.0",
-    "typescript": "^5.1.3"
+    "typescript": "^5.1.3",
+    "validate-branch-name": "^1.3.0"
```
Comment (Alex-2kZharkov) [inline package.json:47]: It is great that you've found a way to do it with some library but we can do it even without any libraries at all. The only thing that we need is a bout of knowledge of `bash` knowledge.

We will add it directly to the `pre-push` hook:
```
BRANCH_NAME=$(git symbolic-ref --short HEAD)
BRANCH_PATTERN="^(feature|fix|test|release)\/HNJ-[0-9]+_[a-zA-Z0-9-]+$"

if [[ ! "$BRANCH_NAME" =~ $BRANCH_PATTERN ]]; then
  echo "Invalid branch name format. Branch name should be in the format of <ticket type>/EVWSNX-<number>_<description>"
  exit 1
else
    echo "Branch name is valid"
fi
```

Comment (Alex-2kZharkov) [reply package.json:47]: Here the `BRANCH_NAME` allows you to get the current branch name. The `BRANCH_PATTERN="^(feature|fix|test|release)\/HNJ-[0-9]+_[a-zA-Z0-9-]+$"` defines the `regex` rule

Then you test it inside `if` statement and throw an with `exit 1` if the branch has incorrect name. The `echo` is like `console.log`. You can feed chat gpt with snippet and ask him to `explain what is it and what does it do`

File: .validate-branch-namerc [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (Alex-2kZharkov) [inline .validate-branch-namerc:1]: Don't forget to remove this file and corresponding package from `package.json` via `npm uninstall` command

**Review by Alex-2kZharkov — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (zarinaazamatova) [reply package.json]: ✅

File: .validate-branch-namerc (hunk 2 above)
Comment (zarinaazamatova) [reply .validate-branch-namerc]: ✅

File: package.json [hunk 3]
```diff
@@ -10,7 +10,8 @@
     "format": "npx prettier --write .",
     "test": "jest",
     "test:ci": "jest --ci",
-    "prepare": "husky install"
+    "prepare": "husky install",
+    "check-branch-name": "chmod 700 .husky/check_branch_naming.sh && sh .husky/check_branch_naming.sh"
```
Comment (Alex-2kZharkov) [inline package.json:14]: please remove this command as well

File: .husky/pre-push [hunk 4]
```diff
@@ -1,4 +1,14 @@
 #!/usr/bin/env sh
 . "$(dirname -- "$0")/_/husky.sh"
 
-npm run test
+BRANCH_NAME=$(git symbolic-ref --short HEAD)
+BRANCH_PATTERN="^(feature|fix|test|release)/HNJ-[0-9]+-[a-zA-Z0-9-]+$"
```
Comment (Alex-2kZharkov) [inline .husky/pre-push:5]: There is a inconsistency. You ask to write branch name like `fix/HNJ-16-improve-pre-push-hook` with `-` between `16` and the next part of description, but the example on line 8 shows `_` is used. Please, select one option and stick to it

**Review by Alex-2kZharkov — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (Alex-2kZharkov) [reply package.json]: good

File: package.json (hunk 3 above)
Comment (zarinaazamatova) [reply package.json]: ✅

File: .husky/pre-push (hunk 4 above)
Comment (zarinaazamatova) [reply .husky/pre-push:5]: ✅

**Review by Alex-2kZharkov — APPROVED** [review]

**Pull request merged**